### PR TITLE
Fix compiler warnings in headers (-Wsign-conversion)

### PR DESCRIPTION
**Pull request by fhuberts**
See #784

### CODE REVIEW

**Comment by fhuberts** [comment]
rebased on #837

File: src/oatpp/web/server/handler/ErrorHandler.cpp [hunk 1]
```diff
@@ -32,8 +32,10 @@ std::shared_ptr<protocol::http::outgoing::Response> ErrorHandler::handleError(co
 
   std::shared_ptr<protocol::http::outgoing::Response> response;
 
+#ifdef __GNUC__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
```
Comment (lganzzzo) [inline src/oatpp/web/server/handler/ErrorHandler.cpp:38]: @fhuberts , I'm not sure if we should hide deprecated methods usage - instead we want it to be fixed.

Comment (fhuberts) [reply src/oatpp/web/server/handler/ErrorHandler.cpp:38]: Yeah I agree, but in one build it fails on the pragmas, and then in another build it also fails when I hid the pragmas.

So I don't know what to do.

Comment (fhuberts) [reply src/oatpp/web/server/handler/ErrorHandler.cpp]: I've pulled out the commit "Wrap GCC pragma in '#ifdef __GNUC__'"

**Review by lganzzzo — APPROVED** [review]

**Comment by lganzzzo** [comment]
merged

**Comment by fhuberts** [comment]
thanks!

**Pull request merged**